py-polars/polars/dataframe/group_by.py

```diff
@@ -307,6 +321,17 @@ def map_groups(self, function: Callable[[DataFrame], DataFrame]) -> DataFrame:
         """
         by: list[str]
 
+        # I don't think any of these checks are necessary as the rule prohibiting
+        # expressions isn't enforced on the lazy side. It should either be better
+        # documented that expressions in the group_by won't be seen by the agg
+        # function or a with_columns can invisibly be inserted before handing off
+        # to the next step to materialize those expressions. In the absence of
+        # direction on this issue, I'll just disallow named_more_by in map_groups
+        # but it is inconsistent with lazy which has no such checks.
+        if len(self.named_more_by) > 0:
+            raise TypeError(
+                "cannot call `map_groups` when grouping with named expressions"
```

Comment:
You can just reuse the message used for the other errors here, it's clear enough. Even better: declare a `msg` variable with this string and re-use it in all 3 error messages.